packages/react-core/src/components/Tooltip/Tooltip.tsx

```diff
@@ -50,150 +49,183 @@ export interface TooltipProps {
+  content: bodyContent,
+  position = 'top',
+  trigger = 'mouseenter focus',
+  isVisible = false,
+  isContentLeftAligned = false,
+  enableFlip = true,
+  className = '',
+  entryDelay = 0,
+  exitDelay = 0,
+  appendTo = () => document.body,
+  zIndex = 9999,
+  maxWidth = tooltipMaxWidth.value,
+  distance = 15,
+  aria = 'describedby',
+  // For every initial starting position, there are 3 escape positions
+  flipBehavior = ['top', 'right', 'bottom', 'left', 'top', 'right', 'bottom'],
+  id = `pf-tooltip-${pfTooltipIdCounter++}`,
+  children,
+  animationDuration = 300,
+  ...rest
+}) => {
+  // could make this a prop in the future (true | false | 'toggle')
+  const hideOnClick = true;
+  const triggerOnMouseenter = trigger.includes('mouseenter');
+  const triggerOnFocus = trigger.includes('focus');
+  const triggerOnClick = trigger.includes('click');
+  const triggerManually = trigger === 'manual';
+  const [visible, setVisible] = React.useState(false);
+  const [opacity, setOpacity] = React.useState(0);
+  const transitionTimerRef = React.useRef(null);
+  const showTimerRef = React.useRef(null);
+  const hideTimerRef = React.useRef(null);
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (!triggerManually) {
+      if (event.keyCode === KEY_CODES.ESCAPE_KEY && visible) {
+        hide();
+      } else if (event.keyCode === KEY_CODES.ENTER && !visible) {
+        show();
+      }
+    }
   };
-
-  handleEscKeyClick = (event: KeyboardEvent) => {
-    if (event.keyCode === KEY_CODES.ESCAPE_KEY && this.tip.state.isVisible) {
-      this.tip.hide();
+  React.useEffect(() => {
+    if (triggerManually) {
+      if (isVisible) {
+        show();
+      } else {
+        hide();
+      }
+    }
+  }, [isVisible, triggerManually]);
+  const show = () => {
+    if (transitionTimerRef.current) {
+      clearTimeout(transitionTimerRef.current);
+    }
+    if (hideTimerRef.current) {
+      clearTimeout(hideTimerRef.current);
     }
+    showTimerRef.current = setTimeout(() => {
+      setVisible(true);
+      setOpacity(1);
+    }, entryDelay);
   };
+  const hide = () => {
+    if (showTimerRef.current) {
+      clearTimeout(showTimerRef.current);
+    }
+    hideTimerRef.current = setTimeout(() => {
+      setOpacity(0);
+      transitionTimerRef.current = setTimeout(() => setVisible(false), animationDuration);
+    }, exitDelay);
+  };
+  const positionModifiers = {
+    top: styles.modifiers.top,
+    bottom: styles.modifiers.bottom,
+    left: styles.modifiers.left,
+    right: styles.modifiers.right
+  };
+  const hasCustomMaxWidth = maxWidth !== tooltipMaxWidth.value;
+  const content = (
+    <div
+      className={css(styles.tooltip, className)}
+      role="tooltip"
+      id={id}
+      style={{
+        maxWidth: hasCustomMaxWidth ? maxWidth : null,
+        opacity,
+        transition: `opacity ${animationDuration}ms cubic-bezier(.54, 1.5, .38, 1.11)`
+      }}
```

Comment:
Can we create a template string or a `getStyles()` function for this since it's shared with `Popover.tsx`?

@mcoker Letting you know this exists.